developers/release.md

```diff
@@ -0,0 +1,209 @@
+
+### Step 9: Update katello-deploy
+
+We maintain a deployment tool, katello-deploy, for easily testing and deploying Katello versions. You'll want to clone the repository (https://github.com/Katello/katello-deploy.git) and update the following:
+
+1. Edit the README to indicate the new version and the OSes it works with.
+1. Update the foreman version matcher - https://github.com/Katello/katello-deploy/blob/master/setup.rb#L10
+1. Add new boxes for X.Y - https://github.com/Katello/katello-deploy/blob/master/Vagrantfile#L28
+
+### Step 10: Test
+
+Use the newly updated katello-deploy to spin up an X.Y box and test for the following:
+
+* Installation
+* UI
+* Repository sync and content view publish
+* Can install katello-agent for el5,6, 7 from the client repo (without the need of the main katello repo enabled)
+* Run a full bats test with content
+
+### Step 11: Branch Documentation
+
+Now we are ready to branch the documentation. This is done separately from the other branching due to the fact that we do not tag or generate an RPM of the documentation website.
+
+```
+git clone https://github.com/Katello/katello.org
+cd katelo.org
+git branch -b KATELLO-X.Y
+git push origin KATELLO-X.Y
+```
+
+Check that the CI job that builds the site (http://ci.theforeman.org/job/deploy_katello_site/) has been kicked off or kick it off manually if needed. Once it passes, visit [http://www.katello.org](http://www.katello.org) and check that X.Y is available from the drop down list. If it did not build correctly, you will need to checkout the `deploy` branch and test it locally.
+
+### Step 12: Announce RC
+
+Now you are ready to announce the RC by sending an email to the foreman-users and foreman-dev mailing lists pointing users at the installation instructions. You should also open a PR to `katello.org` to update the news section with an announcement of the RC release.
+
+### Step 13: Fix Bugs
+
+Start off by creating a label that represents the X.Y version, e.g. `KATELLO-2.1` on the relevant repositories. The workflow until release will be as follows: 
+
+1. Mark any open pull request that should go into X.Y with 'Needs Backporting' and 'Katello X.Y'.
+1. Periodically go through and backport all the merged 'needs backporting' PRs to the X.Y branch and open one new PR against the X.Y branch with all the backports
+1. Remove the 'needs backporting' label from the PRs after cherry picking
+1. After generated 'backport' is merged, tag and release, mash, and copy-repos to fedorahosted
+
+### Step 14: Release!
+
+After iterating through Step 13 a few times over the course of roughly a month, and in sync with Foreman's release date decide that you are ready to release. 
+
+```
+cd katello-misc/repos
+sed -i "s#/gpgcheck=0/#/gpgcheck=1/#g" katello*.repo
+git commit -a -m 'Turning on GPG key checking for X.Y'
+tito tag
+tito release koji
+```
+
+Now sign the RPMs, mash and release.
```

Comment:
Needs some katello.org notes:
  - Backport additional documentation changes, if any
  - Update `latest` in _config.yml on master branch to X.Y